File: 프로그래머스/3/42861. 섬 연결하기/섬 연결하기.py
def solution(n, costs):
    answer = 0
    mat = [[0]*n for _ in range(n)]
    for x, y, c in costs:
        mat[x][y] = c
        mat[y][x] = c
    pre = 10000000
    for i in range(n):
        s = 0
        for  j in range(n):
            s += mat[i][j]
        if pre > s:
            pre = s
        
    return answer

File: 프로그래머스/3/42861. 섬 연결하기/섬 연결하기.java
import java.util.*;
class Solution {
    public static int[] parent;
    public static void init(int n) {
        parent = new int[n];
        for(int i = 0; i<n; i++) {
            parent[i] = i;
        }
    }
    public static int find(int x) {
        if(parent[x] == x)
            return x;
        parent[x] = find(parent[x]);
        return parent[x];
    }
    
    public static void union(int x, int y) {
        int root1 = find(x);
        int root2 = find(y);
        parent[root1] = root2;
    }
    
    public int solution(int n, int[][] costs) {
        int answer = 0;
        init(n);
        Arrays.sort(costs, (o1, o2) -> Integer.compare(o1[2], o2[2]));    
        for(int[] cost : costs) {
            if(find(cost[0]) != find(cost[1])) {
                answer += cost[2];
                union(cost[0], cost[1]);
            }
        }
        return answer;
    }
}